src/commands/up.rs

```diff
@@ -26,15 +26,34 @@ pub struct UpCommand {
     #[clap(short = 'h', long = "help")]
     pub help: bool,
 
+    /// The application to run. This may be a manifest (spin.toml) file, a
+    /// directory containing a spin.toml file, or a remote registry reference.
+    /// If omitted, it defaults to "spin.toml".
+    #[clap(name = "APPLICATION")]
+    pub app_source: Option<String>,
+
+    /// Specifies how to resolve the application argument. The default is 'auto',
+    /// meaning Spin will work it out; you need only specify this if Spin
+    /// is inferring incorrectly.
+    #[clap(
+        name = "APP_SOURCE_TYPE",
+        long = "app-source-type",
+        value_enum,
+        requires = "APPLICATION"
+    )]
+    app_source_type: Option<AppSourceType>,
```

Comment:
You could avoid the option and go with `AppSourceType::Auto` as the default value.